File: spring-web-java/src/main/java/spring/web/java/infrastructure/persistence/adapter/OrderRepositoryAdapter.java
package spring.web.java.infrastructure.persistence.adapter;

import java.util.Optional;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Component;

import lombok.RequiredArgsConstructor;
import spring.web.java.domain.model.entity.Order;
import spring.web.java.domain.model.enums.OrderStatus;
import spring.web.java.domain.repository.OrderRepository;
import spring.web.java.infrastructure.persistence.OrderJpaRepository;
import spring.web.java.infrastructure.persistence.OrderQuerydslRepository;

@Component
@RequiredArgsConstructor
public class OrderRepositoryAdapter implements OrderRepository {

	private final OrderJpaRepository jpaRepository;
	private final OrderQuerydslRepository querydslRepository;

	@Override
	public Page<Order> findAll(Pageable pageable, Long memberId, OrderStatus orderStatus) {
		return querydslRepository.findAll(pageable, memberId, orderStatus);
	}

	@Override
	public Optional<Order> findById(Long id) {
		return jpaRepository.findById(id);
	}

	@Override
	public Order save(Order order) {
		return jpaRepository.save(order);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/infrastructure/persistence/adapter/OrderRepositoryAdapter.kt
package spring.web.kotlin.infrastructure.persistence.adapter

import org.springframework.data.domain.Pageable
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Component
import org.springframework.stereotype.Repository
import spring.web.kotlin.domain.model.entity.Order
import spring.web.kotlin.domain.model.enums.OrderStatus
import spring.web.kotlin.domain.repository.OrderRepository
import spring.web.kotlin.infrastructure.persistence.OrderJpaRepository
import spring.web.kotlin.infrastructure.persistence.OrderQuerydslRepository

@Component
class OrderRepositoryAdapter(
    private val jpaRepository: OrderJpaRepository,
    private val querydslRepository: OrderQuerydslRepository,
) : OrderRepository {
    override fun findAll(pageable: Pageable, memberId: Long?, orderStatus: OrderStatus?) =
        querydslRepository.findAll(pageable, memberId, orderStatus)

    override fun findByIdOrNull(id: Long) = jpaRepository.findByIdOrNull(id)

    override fun save(order: Order) = jpaRepository.save(order)
}